### ISSUE TITLE
cfb_model_artifacts: two_pt_model never splits on `era` (other claims in the original report retracted)

### ISSUE THREAD

**Issue by saiemgilani** [sportsdataverse/sportsdataverse-py#457]
The `two_pt_model.ubj` in the `cfb_model_artifacts` bundle (`model_version` 2026.08.27) is
very nearly a constant, and its level is wrong. It feeds `two_pt_wp` / `two_pt_recommendation`
in both `cfbfastR` and `sportsdataverse-py`, so the two-point decision surface is biased
toward going for two everywhere.

Found while documenting the models for the cfbfastR EP article series
(sportsdataverse/cfbfastR#147). The surface arithmetic is fine — it was verified
bit-identical between the two libraries to eight decimals. The problem is the input model.

## Symptoms

Sweeping each feature with the others held at realistic values:

| Feature | Swept | `prob_2pt` |
|---|---|---|
| `pos_score_diff` | −14 → +14 | 0.487 … **0.578** … 0.557 |
| `posteam_spread` | −28 → +28 | 0.572 … 0.592 |
| `posteam_total` | 35 → 75 | **0.578 at every value** |
| `era` | 0 → 3 | **0.578 at every value** |

Whole output range: **0.487 – 0.592**.

It centres on **0.578** where real college two-point conversion rates run closer to 45% —
and above its own `base_score` of 0.4816, so the trees push *up* from the base rate for
every realistic input.

## Root causes, from the tree dump

1. **`era` is never split on.** It is in the feature list and in none of the 40 trees.
   Two-point rates have moved over 2004–2025; this model cannot express that.

2. **The `posteam_total` splits are at 21.375 and 31.875.** College totals live around
   45–65, so every realistic game takes the same branch — which is why sweeping the total
   changes nothing. Those thresholds look like the training feature was not the game total
   on the scale it is scored with (a scaling or NA-fill problem upstream), rather than a
   modelling choice. **This is the part most worth checking first.**

3. **The first split is `pos_score_diff < -6`.** Nearly every real two-point decision sits
   within a touchdown either way, so the model is flat exactly where it is used.

Gains are correspondingly tiny: `pos_score_diff` 5.06, `posteam_total` 3.38,
`posteam_spread` 1.89.

## Reproduction

No API key needed.

```python
import urllib.request, tempfile, numpy as np, xgboost as xgb
base = ("https://github.com/sportsdataverse/sportsdataverse-data/"
        "releases/download/cfb_model_artifacts/")
r = urllib.request.Request(base + "two_pt_model.ubj", headers={"User-Agent": "Mozilla/5.0"})
p = tempfile.mktemp(suffix=".ubj"); open(p, "wb").write(urllib.request.urlopen(r).read())
m = xgb.Booster(); m.load_model(p)

F = ["posteam_spread", "posteam_total", "pos_score_diff", "era"]
def prob(spread=0, total=55, sd=0, era=3):
    return float(m.predict(xgb.DMatrix(np.array([[spread, total, sd, era]], float),
                                       feature_names=F))[0])

print([round(prob(total=t), 4) for t in (35, 45, 55, 65, 75)])  # all 0.578
print([round(prob(era=e), 4) for e in range(4)])                # all 0.578
print(m.get_score(importance_type="weight"))                    # no 'era' key
print(m.get_dump()[0])                                          # splits at 21.375 / 31.875
```

## Suggested fix

Retrain with the `posteam_total` scaling checked first, and confirm `era` survives into the
trees. The card reports `n_training_rows: 2219971`, which is the full play-by-play frame
rather than the two-point attempt count — worth confirming the label subset is what is
intended, since actual attempts over 2004–2025 should be on the order of thousands.

## Meanwhile

Documented in the cfbfastR series (Part VI) with guidance to read `two_pt_wp_diff` rather
than `two_pt_recommendation`, and to treat small margins as coin flips. Happy to send the
same note to the sdv-py docs if useful.

## Related, lower severity

Two smaller things found in the same sweep, not worth their own issues unless you'd like them:

- **`fg_model`** is non-monotonic at two points (a 49-yard kick rates 0.5882 against 0.5647
  for a 48-yarder; 51 beats 50), and flattens to a constant 0.3656 from a 57-yard kick
  outward. The fourth-down surface differences adjacent yard lines, so the non-monotonicity
  is mildly load-bearing. Era behaviour is healthy (0.556 → 0.675 at 42 yards, era0 → era3).
- **`wp_spread` vs `wp_naive` baseline.** At `spread_time = 0` the two disagree by a
  near-constant −0.08 at every game state. Defensible — a pick'em line for the home team
  implies the visitor is better — but it means `vegas_wp - wp` is not the market's
  contribution, and for a small line late the offset dominates the spread term.

**Comment by saiemgilani** [comment]
## Correction — most of the original report was wrong

I re-checked this against the training code in `cfbfastR-cfb-data`
(`python/cfb_model_build/model_training/features.py`) rather than against the booster
alone. **Three of the four claims above do not hold.** Retracting them here rather than
leaving them to be acted on.

### `posteam_total` is not mis-scaled — I swept the wrong range

`_posteam_total()` is the **possessing team's implied team total**, not the game total:

```python
home_total = (homeTeamSpread + overUnder) / 2
away_total = (overUnder - homeTeamSpread) / 2
```

For a 55-point game with a 7-point spread that is ~24 and ~31. So the real support is
roughly **14–42**, and the splits at **21.375 / 31.875 straddle it correctly**.

My original sweep used 35 → 75, i.e. the *game-total* scale, which is almost entirely
above the feature's support. That is why it looked flat. Over the actual support it is
the model's strongest feature:

| `posteam_total` | `prob_2pt` |
|---|---|
| 14 | 0.4521 |
| 20 | 0.5024 |
| 24 | 0.5213 |
| 30 | 0.5100 |
| 32 | 0.5774 |
| 38 | 0.5780 |

Range 0.126, and 54 of the 106 splits. Nothing upstream is broken.

### The model is not near-constant

Over a realistic grid (spread ±21, team total 16–40, score margin ±10, all eras;
n = 8,580): **mean 0.5114, range 0.365–0.592.** Not 0.487–0.592.

### The level is not "0.578 vs ~45%"

0.578 was a `posteam_total = 55` reading — out of support. At a realistic centre
(total 27, tied, era 3) it predicts **0.518**. And `base_score` is **0.4816**, i.e. the
observed success rate in the training data is **48.2%**, not 45%. The model sits
**+3.0 pp above its own base rate** across the realistic grid, which is what
conditioning is supposed to do.

## What actually stands

**`era` is never split on.** It is in `TWO_PT_FEATURES`, and in none of the 40 trees:

```python
m.get_score(importance_type="weight")
# {'posteam_spread': 20.0, 'posteam_total': 54.0, 'pos_score_diff': 32.0}   # no 'era'
```

Sweeping era 0→3 with everything else fixed moves the prediction by **0.0000**. Two-point
conversion rates have moved across 2004–2025 and this model cannot express that.

With `max_depth=2`, `min_child_weight=40` and 40 rounds over a fairly small attempt
subset, the other three features are simply winning every split. Worth checking whether
that is intended before spending a retrain on it — it may just need more rounds or a
relaxed `min_child_weight`, not a redesign.

## Also worth a look, separately

`NEWS.md` in cfbfastR describes the `xpass` era feature as cutting at
**2006/2013/2017**. The shared constant is `ERA_BOUNDS = (2006, 2013, 2020)`, and both
`_era()` and `_era_onehot()` derive from it — so xpass and FG use **identical cutpoints**
and differ only in encoding (ordinal vs one-hot). The `2017` in `NEWS.md` looks like a
typo; I propagated it into the cfbfastR vignettes and am fixing those.

Retitling this to the `era` finding, which is the only part that survived.
